File: LiveWorks.Core/LiveWorks.Core/Models/Item.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace LiveWorks.Core.Models
{
    public class Item
    {
        [Required]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public double Price { get; set; }
        [Required]
        public string Type { get; set; }
        [Required]
        public double RentalPrice { get; set; }
        public string Description { get; set; }
        public string Dimensions { get; set; }
        public string Supplier { get; set; }
        [Required]
        public int AvailableQuantity { get; set; }
        [Required]
        public DateTime RestockDate { get; set; }
        [Required]
        public int RestockQuantity { get; set; }
    }
}

File: LiveWorks.Core/LiveWorks/ClientApp/EntityDto/item.ts
export class Item {
    public id: number;
    public name: string;
    public price: number;
    public type: string;
    public rentalPrice: number;
    public description: string;
    public dimension: string;
    public supplier: string;
    public availableQuantity: number;
    public restockDate: Date;
    public restockQuantity: number;

    constructor(id: number,
        name: string,
        price: number,
        type: string,
        rentalPrice: number,
        description: string,
        dimension: string,
        supplier: string,
        availableQuantity: number,
        restockDate: Date,
        restockQuantity: number,
    ) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.type = type;
        this.rentalPrice = rentalPrice;
        this.description = description;
        this.dimension = dimension;
        this.supplier = supplier;
        this.availableQuantity = availableQuantity;
        this.restockDate = restockDate;
        this.restockQuantity = restockQuantity;
    }
}